lib/server.js

```diff
@@ -6,29 +6,18 @@ const EventEmitter = require('events')
 const path = require('path')
 const Raven = require('raven')
 
+const channelIsBanned = require('./channel-is-banned-middleware')
+const RedisBus = require('./redis-bus')
```

Comment:
Since this can emit events to the in-memory emitter too, maybe we rename this to `EventBus`? Then we remove any concept at this level of "remote" vs "local". You just emit events, and if Redis is configured it just knows implicitly to use it under the hood. Or is that too "magical"?

Reply:
I'm not opposed 👍 I think that since we don't use the internal event emitter in `lib/server.js`, it makes sense to abstract it properly to this new class. On that note, might even make sense to _define_ the `EventEmitter` on what is now `RedisBus` instead of in `lib/server.js` 🤔

Reply:
> On that note, might even make sense to define the EventEmitter on what is now RedisBus instead of in lib/server.js

Yeh I was thinking along these lines as well.

Reply:
Done 👍